### PR TITLE
Add Parquet export support to OtelTracesSqlEngine

### PR DESCRIPTION
**Pull request by nishanthp**
## Summary
Adds `to_parquet()` method to `OtelTracesSqlEngine` for exporting trace data in Apache Parquet format, enabling efficient storage and high-performance analytics.

## Motivation
Currently, trace data can only be exported to SQL databases or kept in memory as pandas DataFrames. For long-term storage, archival, and sharing trace datasets, a performant columnar file format is needed.

## Changes
- Added `to_parquet()` method with:
  - Configurable compression algorithms (snappy, gzip, brotli, lz4, zstd)
  - Optional partitioning support for efficient filtering
  - Automatic date column extraction for time-based partitioning
- Comprehensive test suite with test cases covering:
  - Basic export functionality
  - Partitioning by service and date
  - Data integrity validation
  - Edge cases (empty dataframes, file size efficiency)

### CODE REVIEW

File: tests/test_instrumentation.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import pandas as pd
```
Comment (AstraBert) [inline tests/test_instrumentation.py]: The PR description says there are 10+ test cases, but here I only see one: is there another test file you did not commit?

**Review by AstraBert — COMMENTED** [review]
The change is ok, but one thing that is not super clear to me is the usefulness: the `to_parquet` method is not used within the Streamlit application: I imagined that you wanted to use it to download the observability data, but in this way it's just an additional method with no direct value whatsoever for the user

**Comment by nishanthp** [comment]
> The change is ok, but one thing that is not super clear to me is the usefulness: the `to_parquet` method is not used within the Streamlit application: I imagined that you wanted to use it to download the observability data, but in this way it's just an additional method with no direct value whatsoever for the user

I wanted to get your opinion on the approach before I could add it to the Streamlit application. 

If the overall approach looks good, I will add the rest of the changes in the next patch.

File: tests/test_instrumentation.py (hunk 1 above)
Comment (nishanthp) [reply tests/test_instrumentation.py]: Yeah, it will be in the next patch.

**Comment by AstraBert** [comment]
The approach looks good, feel free to add it to Streamlit

**Comment by nishanthp** [comment]
Local Testing Snapshots
Screenshot 2026-02-16 at 12 35 27 PM

Screenshot 2026-02-16 at 12 36 56 PM

Screenshot 2026-02-16 at 12 37 53 PM

Screenshot 2026-02-16 at 12 39 48 PM

**Comment by nishanthp** [comment]
Hi @AstraBert,

Just wanted to follow up on this PR. I addressed the earlier feedback and updated the patch.

Please let me know if there are any additional changes needed. I am happy to make further updates if needed. 

Thanks again for the earlier review!

**Review by AstraBert — APPROVED** [review]
